packages/zoe/src/zoeService/authManager.js

```diff
@@ -0,0 +1,113 @@
+// @ts-check
+
+import { MathKind, makeIssuerKit } from '@agoric/ertp';
+import { Far } from '@agoric/marshal';
+
+import './types';
+import './internal-types';
+
+import { makeIssuerStorage } from '../issuerStorage';
+import { makeAndStoreInstanceRecord } from '../instanceRecordStorage';
+import { makeIssuerRecord } from '../issuerRecord';
+import { makeEscrowStorage } from './escrowStorage';
+
+export const makeAuthManager = () => {
+  const issuerStorage = makeIssuerStorage();
+  const escrowStorage = makeEscrowStorage();
+
+  const makeZoeInstanceAuthManager = async (
+    installation,
+    customTerms,
+    uncleanIssuerKeywordRecord,
+  ) => {
+    const { issuers, brands } = await issuerStorage.storeIssuerKeywordRecord(
+      uncleanIssuerKeywordRecord,
+    );
+    Object.entries(issuers).forEach(([keyword, issuer]) =>
+      escrowStorage.createPurse(issuer, brands[keyword]),
```

Comment:
Since a new instance may be using an old issuer and/or brand, and since the issuer-brand correspondence this time may not agree with last time, I keep stumbling here with "Don't we need a check that the current correspondence agrees with the previous correspondence?"

Then I trace it through and become satisfied that we have all the consistency checking we need. These can only disagree if the issuer is misbehaving. If it doesn't misbehave, then `brandIssuerMatch` will always be true and the correspondence will always be the same. If it does misbehave, `brandIssuerMatch` may be true of false. And even if true, the correspondence may be different. However, since that is misbehavior, it is valid for the code here, having verified `brandIssuerMatch` the first time, to simply continue "believing" that first answer and not check later answers.

This line of reasoning deserves an extended comment so the next reader thinking about potential attacks doesn't have to work as hard. Other than inserting explanations, no code changes suggested.

It is worth considering whether this misbehavior may plausibly be accidental rather than malicious. In that case, checking consistency and throwing a diagnostic error would add significant value. I'm on the fence.

Reply:
> since the issuer-brand correspondence this time may not agree with last time

Hmm, I think I'm not seeing why a new instance adds anything to this issue. A malicious issuer and brand could change the answer to whether `brandIssuerMatch` at any time, completely independently of which contract instance the issuer is being used in. 

I'll add a comment in the code about why we only need to check once whether the brand and issuer match. The `brandIssuerMatch` check only prevents a certain kind of malicious behavior - a malicious issuer using another issuer's brand. If a malicious issuer is using another issuer's brand, this check will *never* be true, as you say above. In other words, there can never be a false negative with a valid brand. The only way we get a false negative is if the brand and issuer are both malicious, and then, all bets are off for any contract that uses this issuer. 

I don't think we need to check for accidental misbehavior in Zoe in this regard. It will never occur with an issuer/brand from issuer.js, and we don't have enough data about user patterns to know if this will be an issue otherwise.